tck/src/main/java/org/eclipse/microprofile/lra/tck/participant/api/LraResource.java

```diff
@@ -410,6 +417,34 @@ public Response timeLimit(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId) {
         return Response.status(Response.Status.OK).entity(lraId.toASCIIString()).build();
     }
 
+    /**
+     * @see org.eclipse.microprofile.lra.tck.TckTests#timeLimitWithPreConditionFailed
+     */
+    @GET
+    @Path(TIME_LIMIT_HALF_SEC)
+    @Produces(MediaType.APPLICATION_JSON)
+    @LRA(value = LRA.Type.REQUIRED, timeLimit = 500, timeUnit = ChronoUnit.MILLIS)
+    public Response timeLimitTest2(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId) {
+        assertHeaderPresent(lraId, LRA_HTTP_CONTEXT_HEADER);
+
+        activityStore.add(new Activity(lraId));
+
+        try {
+            Thread.sleep(1000); // sleep for longer than specified in the timeLimit annotation attribute
+            // force the implementation to notice that the LRA should have timed out
+            lraTestService.waitForCallbacks(lraId);
+            // the next request should fail with a 412 code since the LRA should no longer be active
+            restPutInvocation(lraId, MANDATORY_LRA_RESOURCE_PATH, "");
+        } catch (WebApplicationException wae) {
+            return wae.getResponse();
```

Comment:
can you also log the exception or rethrow it? With the current code, the test fails only on failed assertion on status. The error message can be helpful.

Reply:
It is only an error if the status is not 412.
412 is the expected status and the resource is returning that to the test. The test has the responsibility to do the assertion. Add a log message here would at best provide tracing and we don't normally trace TCK tests. The calling test will report any unexpected status code.

Reply:
sorry, I forgot that this method throws an exception when the status is different than 200.